Packs/ExabeamDataLake/Integrations/ExabeamDataLake/ExabeamDataLake.py

```diff
@@ -0,0 +1,273 @@
+        """
+        headers = {"Csrf-Token": "nocheck"}
+        data = {"username": self.username, "password": self.password}
+
+        self._http_request(
+            "POST",
+            full_url=f"{self._base_url}/api/auth/login",
+            headers=headers,
+            data=data,
+        )
+
+    def test_module_request(self):
+        """
+        Performs basic get request to check if the server is reachable.
+        """
+        self._http_request('GET', full_url=f'{self._base_url}/api/auth/check', resp_type='text')
+
+    def query_datalake_request(self, search_query: dict) -> dict:
+        return self._http_request(
+            "POST",
+            full_url=f"{self._base_url}/dl/api/es/search",
+            data=json.dumps(search_query),
+            headers={"Content-Type": "application/json"},
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def _parse_entry(entry: dict) -> dict:
+    """
+    Parse a single entry from the API response to a dictionary.
+
+    Args:
+        entry: The entry from the API response.
+
+    Returns:
+        dict: The parsed entry dictionary.
+    """
+    source: dict = entry.get("_source", {})
+    return {
+        "Id": entry.get("_id"),
+        "Vendor": source.get("Vendor"),
+        "Created_at": source.get("@timestamp"),
+        "Product": source.get("Product"),
+        "Message": source.get("message")
+    }
+
+
+def dates_in_range(start_time: Any, end_time: Any) -> list[str]:
+    """
+     Generate a list of dates within a specified range.
+
+    Args:
+        start_time: The start date of the range in the format "YYYY-MM-DD".
+        end_time: The end date of the range in the format "YYYY-MM-DD".
+
+    Raises:
+        DemistoException: If the start time is not before the end time, or if the difference between start time and end time is
+        greater than 10 days.
+
+    Returns:
+        list: A list of dates within the specified range, formatted as strings in the format "YYYY.MM.DD".
+    """
+    start_time = datetime.strptime(start_time, "%Y-%m-%d")
+    end_time = datetime.strptime(end_time, "%Y-%m-%d")
+
+    if start_time >= end_time:
+        raise DemistoException("Start time must be before end time")
+
+    if (end_time - start_time).days > 10:
+        raise DemistoException("Difference between start time and end time must be less than or equal to 10 days")
+
+    dates = []
+    current_date = start_time
+    while current_date <= end_time:
+        dates.append(current_date.strftime("%Y.%m.%d"))
+        current_date += timedelta(days=1)
+
+    return dates
+
+
+def get_date(time: str):
+    """
+    Get the date from a given time string.
+
+    Args:
+        time (str): The time string to extract the date from.
+
+    Returns:
+        str: The date extracted from the time string formatted in ISO 8601 format (YYYY-MM-DD),
+        or None if the time string is invalid.
+    """
+    date_time = arg_to_datetime(arg=time, arg_name="Start time", required=True)
```

Comment:
I know that with required True this will return an exception if the arg is missing or there is an issue. 
But its no a best practice to define a variable inside inside if and then return it where its not defined in the outer scope.
Why do you need this if?

Reply:
Because strftime doesn't get it if we don't use this if.